.github/workflows/merge.yml

```diff
@@ -1,34 +1,43 @@
-name: Merge Build
+name: Merge Build and Push to Docker Hub
 
 on:
   push:
     branches:
       - develop
 
 jobs:
-  build-docker-images:
+  build-backend:
     runs-on: ubuntu-latest
-
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Set up Node.js
-        uses: actions/setup-node@v3
-        with:
-          node-version: '20'
+      - name: Build Backend Docker Image
+        working-directory: ./backend
+        run: |
+          docker build -t petclinic-backend:latest .
+          docker tag petclinic-backend:latest ahmedhamdo/petclinic-backend:latest
+
+      - name: Push Backend Docker Image to Docker Hub
```

Comment:
There is a github action for build&push.
https://github.com/docker/build-push-action